app/scripts/controllers/eventRegistrations.js

```diff
@@ -563,59 +632,82 @@ angular
+        );
+        const { title, yesString, warningMessage } =
+          $scope.buildDeletionWarningMessage(isCouple);
 
-        modalMessage
-          .confirm({
-            title: 'Delete Registration',
-            question:
-              'Are you sure you want to delete this registration?<br>There is no recovering the data once deleted.',
-            yesString: 'Delete',
-            noString: 'Cancel',
-            normalSize: true,
-          })
+        $scope
+          .showModal(title, warningMessage, yesString)
           .then(function () {
-            $http
-              .get('registrations/' + registrant.registrationId)
-              .then(function (response) {
-                var registration = response.data;
-                var url = 'registrations/' + registration.id;
-
-                if (registration.registrants.length > 1) {
-                  //Delete Registrant
-                  url = 'registrants/' + registrant.id;
-                }
-
-                $http({
-                  method: 'DELETE',
-                  url: url,
-                })
-                  .then(function () {
-                    _.remove($scope.registrants, function (r) {
-                      return r.id === registrant.id;
-                    });
-                    var reg = $scope.getRegistration(registrant.registrationId);
-                    if (angular.isDefined(reg)) {
-                      _.remove(reg.registrants, function (r) {
-                        return r.id === registrant.id;
-                      });
-                      _.remove(reg.groupRegistrants, function (r) {
-                        return r.id === registrant.id;
-                      });
-                    }
-                  })
-                  .catch(function (response) {
-                    modalMessage.error({
-                      message:
-                        response.data && response.data.error
-                          ? response.data.error.message
-                          : 'An error has occurred while deleting this registration.',
-                    });
-                  });
-              });
+            return $http.get('registrations/' + registrant.registrationId);
+          })
+          .then(function (response) {
+            const registration = response.data;
+
+            let registrantsToDelete = [registrant];
+            if (isCouple) {
+              registrantsToDelete = $scope.findCoupleRegistrants(
+                registrant,
+                registration,
+              );
+            }
+
+            const reg = $scope.getRegistration(registrant.registrationId);
+            let url;
+            if (reg.registrants.length === 1 || isCouple) {
+              url = 'registrations/' + registration.id;
+            } else {
+              url = 'registrants/' + registrant.id;
+            }
+            $http.delete(url).then(function () {
+              $scope.refreshRegistrations();
```

Comment:
[Must-fix] Missing error handling in HTTP delete operation. Evidence: `$http.delete(url).then(...)` without .catch(). Impact: Failed deletions could leave UI in inconsistent state. Fix: Add .catch() handler. [Category: File]